examples/structured_data/classification_with_tfdf.py

```diff
@@ -0,0 +1,726 @@
+)
+test_data["income_level"] = test_data["income_level"].apply(
+    lambda x: 0 if x == " - 50000." else 1
+)
+
+"""
+Now let's show the shapes of the training and test dataframes, and display some instances.
+"""
+
+print(f"Train data shape: {train_data.shape}")
+print(f"Test data shape: {test_data.shape}")
+train_data.head().T
+
+"""
+## Define dataset metadata
+
+Here, we define the metadata of the dataset that will be useful for encoding
+the input features with respect to their types.
+"""
+
+# Target feature name.
+TARGET_FEATURE_NAME = "income_level"
+# Weight column name.
+WEIGHT_COLUMN_NAME = "instance_weight"
+# Numeric feature names.
+NUMERIC_FEATURE_NAMES = [
+    "age",
+    "wage_per_hour",
+    "capital_gains",
+    "capital_losses",
+    "dividends_from_stocks",
+    "num_persons_worked_for_employer",
+    "weeks_worked_in_year",
+]
+# Categorical features and their vocabulary lists.
+CATEGORICAL_FEATURES_WITH_VOCABULARY = {
+    feature_name: sorted(
+        [str(value) for value in list(train_data[feature_name].unique())]
+    )
+    for feature_name in CSV_HEADER
+    if feature_name
+    not in list(NUMERIC_FEATURE_NAMES + [WEIGHT_COLUMN_NAME, TARGET_FEATURE_NAME])
+}
+# All features names.
+FEATURE_NAMES = NUMERIC_FEATURE_NAMES + list(
+    CATEGORICAL_FEATURES_WITH_VOCABULARY.keys()
+)
+# Feature default values.
+COLUMN_DEFAULTS = [
+    [0.0]
+    if feature_name in NUMERIC_FEATURE_NAMES + [TARGET_FEATURE_NAME, WEIGHT_COLUMN_NAME]
+    else ["NA"]
+    for feature_name in CSV_HEADER
+]
+
+"""
+## Configure hyperparameters
+
+You can find all the parameters of the Gradient Boosted Tree model in the
+[documentation](https://www.tensorflow.org/decision_forests/api_docs/python/tfdf/keras/GradientBoostedTreesModel)
+"""
+
+GROWING_STRATEGY = "BEST_FIRST_GLOBAL"
+NUM_TREES = 250
+MIN_EXAMPLES = 6
+MAX_DEPTH = 5
+SUBSAMPLE = 0.65
+SAMPLING_METHOD = "RANDOM"
+VALIDATION_RATIO = 0.1
+
+"""
+## Implement a training and evaluation procedure
+"""
+
+
+def fix_datatypes(features):
+    for feature_name in features:
+        if feature_name in CATEGORICAL_FEATURES_WITH_VOCABULARY:
+            if features[feature_name].dtype != tf.dtypes.string:
+                # Convert categorical feature values to string.
+                features[feature_name] = tf.strings.as_string(features[feature_name])
+    return features
+
+
+def run_experiment(model, train_data, test_data, num_epochs=1, batch_size=None):
+
+    train_dataset = tfdf.keras.pd_dataframe_to_tf_dataset(
+        train_data, label=TARGET_FEATURE_NAME, weight=WEIGHT_COLUMN_NAME
+    ).map(lambda features, target, weight: (fix_datatypes(features), target, weight))
+    test_dataset = tfdf.keras.pd_dataframe_to_tf_dataset(
+        test_data, label=TARGET_FEATURE_NAME, weight=WEIGHT_COLUMN_NAME
+    ).map(lambda features, target, weight: (fix_datatypes(features), target, weight))
+
+    history = model.fit(train_dataset, epochs=num_epochs, batch_size=batch_size)
```

Comment:
No need for num_epoch and batch_size.

You can add comment that the algorithm should be trained for exactly one epoch and that the batch size does not matter.